src/main/java/net/mcreator/element/types/Armor.java

```diff
@@ -44,21 +46,21 @@
 @SuppressWarnings("unused") public class Armor extends GeneratableElement implements IItem, ITabContainedElement {
 
 	public boolean enableHelmet;
-	public String textureHelmet;
+	@TextureReference(TextureType.ITEM) public String textureHelmet;
 	public boolean enableBody;
-	public String textureBody;
+	@TextureReference(TextureType.ITEM) public String textureBody;
 	public boolean enableLeggings;
-	public String textureLeggings;
+	@TextureReference(TextureType.ITEM) public String textureLeggings;
 	public boolean enableBoots;
-	public String textureBoots;
+	@TextureReference(TextureType.ITEM) public String textureBoots;
 
 	public Procedure onHelmetTick;
 	public Procedure onBodyTick;
 	public Procedure onLeggingsTick;
 	public Procedure onBootsTick;
 
 	public TabEntry creativeTab;
-	public String armorTextureFile;
+	@TextureReference(TextureType.ARMOR) public String armorTextureFile;

```

Comment:
This one is not working as it requires layer_1 or layer_2 in the texture name additionally

Reply:
A `String[] files = "%s"` parameter would resolve that?

Reply:
That would allow to specify templates for actual file names that the value of the target field should be parsed into (e.g. with value `files = { "%s_layer_1", "%s_layer_2" }` a string `"test"` would result into files `test_layer_1` and `test_layer_2`).

Reply:
Ok so parameter in annotation, yep, that can work :)

Reply:
:heavy_check_mark: Fixed